### PR TITLE
New snapshot lkm support

### PR DESCRIPTION
**Pull request by vanhauser-thc**

### CODE REVIEW

File: include/snapshot-inl.h [hunk 1]
```diff
@@ -35,25 +34,82 @@
...
+struct afl_snapshot_vmrange_args {
+
+  unsigned long start, end;
+
+};
 
 static int afl_snapshot_dev_fd;
 
-static int afl_snapshot_init(void) {
+static int afl_snapshot_init() {
```
Comment (domenukk) [inline include/snapshot-inl.h:69]: Does this need variable args?

Comment (vanhauser-thc) [reply include/snapshot-inl.h:69]: no args to that function

Comment (domenukk) [reply include/snapshot-inl.h:69]: Then we shoudn't drop the `void` here :)

Comment (vanhauser-thc) [reply include/snapshot-inl.h:69]: why? AFAIK () means no parameter, (void) means ... I dont know :) same would be my guess

**Pull request merged**